src/Engine/IO/Window.cpp

```diff
@@ -20,30 +19,41 @@ void Window::errorCallback(int error, const char *desc) {
 void Window::keyCallback(GLFWwindow *window, int key, int scancode, int action, int mods) {
     if (key == GLFW_KEY_ESCAPE && action == GLFW_PRESS) {
         glfwSetWindowShouldClose(window, true);
+        return;
     }
 
 #ifdef DEBUG_MODE
-    if (isImGuiEnabled() && (ImGui::IsWindowFocused() || ImGui::IsMouseHoveringAnyWindow())) {
+    if (key == GLFW_KEY_GRAVE_ACCENT && action == GLFW_PRESS) {
+        toggleImGui();
+    }
+    else if (isImGuiEnabled() && (ImGui::IsWindowFocused() || ImGui::IsMouseHoveringAnyWindow())) {
         ImGui_ImplGlfwGL3_KeyCallback(window, key, scancode, action, mods);
     }
     else
 #endif
     {
-        Keyboard::setKeyStatus(key, action);
-        Scene::sendMessage<KeyMessage>(nullptr, key, action, mods);
+        if (!cursorEnabled) {
+            Keyboard::setKeyStatus(key, action);
+            Scene::sendMessage<KeyMessage>(nullptr, key, action, mods);
+        }
     }
 }
 
 void Window::mouseButtonCallback(GLFWwindow *window, int button, int action, int mods) {
 #ifdef DEBUG_MODE
-    if (isImGuiEnabled() && (ImGui::IsWindowFocused() || ImGui::IsMouseHoveringAnyWindow())) {
+    if (button == GLFW_MOUSE_BUTTON_3 && action == GLFW_PRESS && isImGuiEnabled()) {
+        toggleCursorEnabled();
+    }
+    else if (isImGuiEnabled() && (ImGui::IsWindowFocused() || ImGui::IsMouseHoveringAnyWindow())) {
```

Comment:
Same here

Reply:
same deal, the event should be "consumed" and not further propagate into the system